pharo/run.sh

```diff
@@ -87,9 +93,45 @@ lastSuccessfulBuild/artifact/${moose_name}.zip"
 # Return:
 #   Pharo vm download url
 ################################################################################
+
+# get os
+pharo::get_os() {
+	local TMP_OS=`uname | tr "[:upper:]" "[:lower:]"`
+	if [[ "{$TMP_OS}" = *darwin* ]]; then
+	    echo "mac";
+	elif [[ "{$TMP_OS}" = *linux* ]]; then
+	    echo "linux";
+	elif [[ "{$TMP_OS}" = *win* ]]; then
+	    echo "win";
+	elif [[ "{$TMP_OS}" = *mingw* ]]; then
+	    echo "win";
+	else
+	    echo "Unsupported OS";
+	    exit 1;
+	fi
+}
+
+# get vm url
+# variables: 
+#  PHARO_VM=stable*|latest
+#  LINUX_HEARTBEAT=threaded*|itimer
 pharo::get_vm_url() {
   local smalltalk_name=$1
+  local os="$(pharo::get_os)"
+  local heartbeat=""
+  local latest=""
 
+  if [ "${PHARO_VM}" == "latest" ]; then
```

Comment:
Needs to be `${PHARO_VM:-}`, otherwise you get the `PHARO_VM: unbound variable` error.